CredentialProvider.Microsoft/CredentialProviders/Vsts/BearerTokenProviders.cs

```diff
@@ -1,153 +1,155 @@
-    {
-        private readonly IAdalTokenProvider adalTokenProvider;
-
-        public AdalCacheBearerTokenProvider(IAdalTokenProvider adalTokenProvider)
-        {
-            this.adalTokenProvider = adalTokenProvider;
-        }
-
-        public bool Interactive { get; } = false;
-        public string Name { get; } = "ADAL Cache";
-
-        public async Task<string> GetTokenAsync(Uri uri, CancellationToken cancellationToken)
-        {
-            return (await adalTokenProvider.AcquireTokenSilentlyAsync(cancellationToken))?.AccessToken;
-        }
-
-        public bool ShouldRun(bool isRetry, bool isNonInteractive, bool canShowDialog)
-        {
-            return !isRetry;
-        }
-    }
-
-    /// <summary>
-    /// Acquire an AAD token via Windows Integrated Authentication
-    /// </summary>
-    public class WindowsIntegratedAuthBearerTokenProvider : IBearerTokenProvider
-    {
-        private readonly IAdalTokenProvider adalTokenProvider;
-
-        public WindowsIntegratedAuthBearerTokenProvider(IAdalTokenProvider adalTokenProvider)
-        {
-            this.adalTokenProvider = adalTokenProvider;
-        }
-
-        public bool Interactive { get; } = false;
-        public string Name { get; } = "ADAL Windows Integrated Authentication";
-
-        public async Task<string> GetTokenAsync(Uri uri, CancellationToken cancellationToken)
-        {
-            return (await adalTokenProvider.AcquireTokenWithWindowsIntegratedAuth(cancellationToken))?.AccessToken;
-        }
-
-        public bool ShouldRun(bool isRetry, bool isNonInteractive, bool canShowDialog)
-        {
-            return WindowsIntegratedAuthUtils.SupportsWindowsIntegratedAuth() && EnvUtil.WindowsIntegratedAuthenticationEnabled();
-        }
-    }
-
-    /// <summary>
-    /// Acquire an AAD token by showing a sign-in UI
-    /// </summary>
-    public class UserInterfaceBearerTokenProvider : IBearerTokenProvider
-    {
-        private readonly IAdalTokenProvider adalTokenProvider;
-        private readonly ILogger logger;
-
-        public UserInterfaceBearerTokenProvider(IAdalTokenProvider adalTokenProvider, ILogger logger)
-        {
+﻿// Copyright (c) Microsoft. All rights reserved.
+//
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using NuGetCredentialProvider.Logging;
+using NuGetCredentialProvider.Util;
+
+namespace NuGetCredentialProvider.CredentialProviders.Vsts
+{
+    /// <summary>
+    /// Acquire an AAD token via the ADAL cache
+    /// </summary>
+    public class AdalCacheBearerTokenProvider : IBearerTokenProvider
+    {
+        private readonly IAdalTokenProvider adalTokenProvider;
+
+        public AdalCacheBearerTokenProvider(IAdalTokenProvider adalTokenProvider)
+        {
             this.adalTokenProvider = adalTokenProvider;

```

Comment:
VS complained about line endings :(

Reply:
The real changes in this file are:
- added isOnPrem to ShouldRun
- WindowsIntegratedAuthBearerTokenProvider  doesn't run on prem